libs/execution_context/ebpf_maps.c

```diff
@@ -1063,6 +1070,80 @@ _update_queue_map_entry(
     return result;
 }
 
+static ebpf_core_map_t*
+_create_stack_map(_In_ const ebpf_map_definition_in_memory_t* map_definition)
+{
+    size_t stack_map_size =
+        EBPF_OFFSET_OF(ebpf_core_stack_map_t, slots) + map_definition->max_entries * sizeof(uint8_t*);
```

Comment:
why sizeof(uint8_t*) instead of sizeof(map_definition->value_size)?
I'm guessing you do dynamic allocation of values just in case they're large and the stack depth is much less than max_entries.  At least add a comment about this, but I'm not sure it's a good tradeoff.   Just like BPF_MAP_TYPE_ARRAY uses embedded values, this could too for simplicity.

Reply:
It's required for correctness. Assume the following sequence:

1) CPU0: bpf_map_update_elem
2) CPU0: bpf_map_lookup_and_delete_elem -> pops entry of stack.
3) CPU1: bpf_map_update_elem
4) CPU1: bpf_map_lookup_and_delete_elem -> pops entry of stack.

CPU0 and CPU1 are now sharing the memory, which is probably not what was intended.

Reply:
Got it, please add comment explaining that.  And probably add it to BPF_MAP_TYPE_QUEUE as well.